app/src/app-logic.js

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import { useAragonApi, useConnectedAccount } from "@aragon/api-react";
+import usePanelState from './hooks/usePanelState'
+import useUnwrapPanelState from './hooks/unwrapPanelState'
+
+function noop() {}
+
+// Unwrap tokens action
+export function useUnwrapTokensAction(onDone = noop) {
+  const { api, appState } = useAragonApi();
+  return useCallback(
+    async amount => {
+      await api.unlock(amount).toPromise();
+      onDone()
+    },
+    [api, appState]
```

Comment:
`onDone` should be listed in the dependencies:

```suggestion
    [api, onDone]
```

We should probably add ESLint and [eslint-plugin-react-hooks](https://www.npmjs.com/package/eslint-plugin-react-hooks), it makes these much easier to spot 😅